### ISSUE TITLE
MetadataSet.class.php

### ISSUE THREAD

**Issue by marknokes** [wingmingchan/php7-cascade-ws-ns#3]
Hi, Wing!

I'm using your sync example, and it's fantastic! I was getting a null service exception when running $instances->updateMetadatasetContainer() and noticed that the MetadataSet.class.php called for a new DynamicMetadataFieldDefinition in two places where the service wasn't being passed in. I've attached the updated code. Thanks again for all you do!
[MetadataSet.class.txt](https://github.com/wingmingchan/php7-cascade-ws-ns/files/1876084/MetadataSet.class.txt)

**Comment by wingmingchan** [comment]
$(UI updated the code on 1/22/2018. https://github.com/wingmingchan/php7-cascade-ws-ns/blob/master/cascade_ws_ns7/asset_classes/MetadataSet.class.php You are still using an older version. 

Wing

>>> Mark Nokes <notifications@github.com> 4/4/2018 10:06 AM >>>

Hi, Wing! 
I'm using your sync example, and it's fantastic! I was getting a null service exception when running $instances->updateMetadatasetContainer() and noticed that the MetadataSet.class.php called for a new DynamicMetadataFieldDefinition in two places where the service wasn't being passed in. I've attached the updated code. Thanks again for all you do!
MetadataSet.class.txt ( https://github.com/wingmingchan/php7-cascade-ws-ns/files/1876084/MetadataSet.class.txt )

**Comment by wingmingchan** [comment]
$(UI forgot to ask this. Are you trying to synch between two Cascade instances? 

Wing

>>> Mark Nokes <notifications@github.com> 4/4/2018 10:06 AM >>>

Hi, Wing! 
I'm using your sync example, and it's fantastic! I was getting a null service exception when running $instances->updateMetadatasetContainer() and noticed that the MetadataSet.class.php called for a new DynamicMetadataFieldDefinition in two places where the service wasn't being passed in. I've attached the updated code. Thanks again for all you do!
MetadataSet.class.txt ( https://github.com/wingmingchan/php7-cascade-ws-ns/files/1876084/MetadataSet.class.txt )

**Comment by marknokes** [comment]
I used the version in the 8.8 folder. Yes, I’m syncing between two instances.

**Comment by wingmingchan** [comment]
$(UThat version is a couple of months behind. Use the one named cascade_ws_ns7. This newest version provides both REST and SOAP. It also provides an $admin object that does all, and then some, that $cascade can do. I have not updated CascadeInstances though. 

You are very brave to have decided to use the library to sync assets. I believe that you are the first one who dares to try. 

Wing 


>>> Mark Nokes <notifications@github.com> 4/4/2018 2:15 PM >>>

I used the version in the 8.8 folder. Yes, I’m syncing between two instances.

**Comment by marknokes** [comment]
I see. Thanks for letting me know. Lol, I must be brave! Well, I've developing in my test instance, and thought...I'll bet Wing created something to synchronize assets...and behold! You did!!! So far I have copied everything except workflows (because I don't have any to copy), transports, and destinations. It's working really great.

**Comment by wingmingchan** [comment]
$(UBack in 2013, I tried to write a program to sync assets across two Cascade instances and failed miserably. That's how I started building my library. And I told the audience in the 2014 Cascade conference that this was the ultimate test of any web service library. Back then, I said I would give the community three years to catch up with me. It has taken more than three years for you to try. 

I also said that I would give the community three years to catch up with me with respect to Velocity two years ago. I am still waiting. I want to talk about trees this year.

>>> Mark Nokes <notifications@github.com> 4/4/2018 3:50 PM >>>

I see. Thanks for letting me know. Lol, I must be brave! Well, I've developing in my test instance, and thought...I'll be Wing created something to synchronize assets...and behold! You did!!! So far I have copied everything except workflows (because I don't have any to copy), transports, and destinations. It's working really great.
